main.cc

```diff
@@ -11,49 +15,104 @@
+    std::cout << "Final Score:" << std::endl;
+    std::cout << "White: " << scoreBoard.whitePoints << std::endl;
+    std::cout << "Black: " << scoreBoard.blackPoints << std::endl;
+}
 
-    while (true) {
-        int blackWins = 0;
-        int whiteWins = 0;
 
-        // run the helper function here
+void buildBoard(std::vector<std::vector<Piece *>> &board) {
+    for (int row = 0; row < 8; ++row) {
+        std::vector<Piece *> boardRow;
+        for (int col = 0; col < 8; ++col) { 
+            boardRow.push_back(nullptr);        //initializing all places to nullptrs
+        } 
+        board.push_back(boardRow);
+    } 
+}
 
-        // cout the number of wins here
-    }
-    
 
+void gameInstance() {
+
+    //iterate over vector of vectors and build base board - possibility of keeping in main
+    std::vector<std::vector<Piece *>> ChessBoard;
+    buildBoard(ChessBoard);
+    std::vector<std::vector<Piece *>> emptyBoard;
+    buildBoard(emptyBoard);
 
-}
\ No newline at end of file
+    ChessGame game{&ChessBoard};        //constructor takes in ptr to board
+    std::vector<Piece *> whitePieces;
+    std::vector<Piece *> blackPieces;
+
+    bool setupFlag = 0;         //changed to 1 if "setup" command executed (must be successful, else cannot leave mode)
+    bool gameRunFlag = 0;       //changed to 1 if "game" command executed and when gameRun finishes
+    std::string comm = "";
+    while (!(gameRunFlag) && (cin >> comm)) {
+        if (comm == "setup") {
+            if (setupFlag) {
+                ChessBoard.clear();
+                buildBoard(ChessBoard);
+                game.setBoard(&ChessBoard);    //create method to take in pointer to the board and change board in class
+            }
+            userSetup(&game, whitePieces, blackPieces);     //takes references to the vectors and changes them
+            setupFlag = 1;
+        } else if (comm == "game") {
+            std::string wPlayer = "";
+            std::string bPlayer = "";
+            std::cin >> wPlayer >> bPlayer;
+
+            //later, constructPlayer(player) will be defined to construct human/l1/l2/l3/l4
+            //Human for now, till level subclasses are created and defined
+            Human whitePlayer{&game, "human"};  //+ &whitePieces for computer
+            Human blackPlayer{&game, "human"};  //+ &blackPieces for computer
+
+            if (!(setupFlag)) {
+                game.setBoard(&emptyBoard);
+                defaultSetup(&game, whitePieces, blackPieces);
+            }
+
+            int whoWon = gameRun(&whitePlayer, &blackPlayer, &game);
+            if (whoWon == 1) {
+                tracker.whitePoints += 1;
+            } else if (whoWon = -1) {
+                tracker.blackPoints += 1;
+            } else if (whoWon == 0) {
+                tracker.whitePoints += 0.5;
+                tracker.blackPoints += 0.5;
+            } else {
+                std::cout << "Game left unfinished!" << std::endl;
+            }
```

Comment:
I had a comment in the `gamerun.cc` file asking whether any other value aside from 0, 1 or -1 can be returned. If it is the case where only 0, 1 or -1 is returned, how does the `gameRun` function return a value that reaches this specific `else` statement?

If it isn't the case, what are the conditions for the game to be "left unfinished'?

Reply:
Responded to in the previous comment.